internal/assets/public/public.go

```diff
@@ -0,0 +1,109 @@
+package public
```

Comment:
This file seems having more complicated logic than I would expect, do you think there is a way to reuse some stdlib functions/methods to simplify it? Or did you copy it from go-bindata/go-bindata?

Reply:
yes, it's is copy from go-bindata/go-bindata that base [PR](https://github.com/go-bindata/go-bindata/pull/18) and make a lite optimize just do it fun. Maybe have some other way optimize it take a time later :)